app/assets/javascripts/analytics/_events.js

```diff
@@ -88,13 +88,31 @@
 
   GOVUK.GDM.analytics.events = {
     'supplierListDownload': function (e) {
-      var linkClick = new LinkClick(e);
+      var currentURL = GOVUK.GDM.analytics.location.pathname();
```

Comment:
you should use:

`var currentURL = window.location.pathname;`

Reply:
Could you maybe explain why, please? It would be much more useful if you did that, thanks!

Reply:
I'm curious too - we use the `GOVUK.GDM.analytics.location` in several places here and in the briefs FE. There must be some benefit (filtering out test events maybe)?

Reply:
The reason why I would suggest using the native location functions is because you would always get a consistent response. If someone decides to change the function to return something other than what you expect, it would cause your function to break. 

Also this it's just a wrapper class unless you want to modify the value returned. I would recommend just using the native functionality. There are no additional benefits that I can see from the function below. If you want to continue to use it due to the consistency it's fine I guess.

``` 
(function (root) {
  "use strict";

  root.GOVUK.GDM = root.GOVUK.GDM || {};
  root.GOVUK.GDM.analytics = {
    'register': function () {
      var cookieDomain = (root.document.domain === 'www.digitalmarketplace.service.gov.uk') ? '.digitalmarketplace.service.gov.uk' : root.document.domain;
      var universalId = 'UA-49258698-1';

      GOVUK.Analytics.load();
      GOVUK.analytics = new GOVUK.Analytics({
        universalId: universalId,
        cookieDomain: cookieDomain
      });
    },

    // wrapper around access to window.location
    'location': {
      'hostname': function () {
        return root.location.hostname;
      },
      'pathname': function () {
        return root.location.pathname;
      },
      'protocol': function () {
        return root.location.protocol;
      },
      'search': function () {
        return root.location.search;
      }
    }
  };
})(window);
```